idx/memory/memory.go

```diff
@@ -352,31 +362,20 @@ func (m *MemoryIdx) indexTags(def *schema.MetricDefinition) {
 // a return value of "false" means there was an error and the deindexing was
 // unsuccessful, "true" means the indexing was at least partially or completely
 // successful
-func (m *MemoryIdx) deindexTags(tags TagIndex, def *schema.MetricDefinition) bool {
+func (m *MemoryIdx) deindexTags(tags TagIndex, def *idx.MetricDefinition) bool {
 	for _, tag := range def.Tags {
-		tagSplits := strings.SplitN(tag, "=", 2)
-		if len(tagSplits) < 2 {
-			// should never happen because every tag in the index
-			// must have a valid format
-			invalidTag.Inc()
-			log.Errorf("memory-idx: Tag %q of id %q has an invalid format", tag, def.Id)
-			continue
-		}
-
-		tagName := tagSplits[0]
-		tagValue := tagSplits[1]
-		tags.delTagId(tagName, tagValue, def.Id)
+		tags.delTagId(tag.Key, tag.Value, def.Id, m)
```

Comment:
I personally love that we stop storing tags in the format `key=value`, but as separate strings instead. That way we get rid of the repetitive splitting by `=` with error checks.